app/helpers/sufia_helper.rb

```diff
@@ -15,4 +15,19 @@ def link_to_fields(fields, field_string)
     fields.map { |tag| link_to_facet(tag, field_string) }
   end
 
+  def link_to_field(fieldname, fieldvalue, displayvalue = nil)
+    if fieldvalue["id"]
+      p = { search_field: 'advanced', fieldname => '"' + fieldvalue["id"] + '"' }
+    else
+      p = { search_field: 'advanced', fieldname => '"' + fieldvalue + '"' }
+    end
+    link_url = catalog_index_path(p)
+    if fieldvalue["preflabel"]
+      display = displayvalue.blank? ? fieldvalue["preflabel"] : displayvalue
+    else
+      display = displayvalue.blank? ? fieldvalue : displayvalue
+    end
+    display = displayvalue.blank? ? fieldvalue : displayvalue
```

Comment:
Is this a duplicate of 28?  Seems like you're doing a test then overwriting the result.